### PR TITLE
Improve README structure and clarity

### PR DESCRIPTION
**Pull request by seekersoftec**
### Objective

Improve the README structure and clarity for better user experience and navigation. 

### Changes
* Introduced a detailed table of contents with descriptive headings.
* Enhanced section headings using emojis for visual clarity.
* Reorganized sections for improved logical flow.

### Scope of Change
This change primarily affects the README file, improving its readability and overall structure.

Solves [#7](https://github.com/Bitshala-Incubator/silent-pay-indexer/issues/7)

### CODE REVIEW

**Comment by theanmolsharma** [comment]
Let's add an experimental warning to the README. You can take a look at [here](https://github.com/Bitshala-Incubator/silent-pay) for reference.
Additionally, let's squash all commits into one.

**Comment by seekersoftec** [comment]
Okay, will take a look

File: README.md [hunk 1]
```diff
@@ -1,73 +1,189 @@
...
-
-## Description
-
-[Nest](https://github.com/nestjs/nest) framework TypeScript starter repository.
-
-## Installation
+# Silent Payment Indexer
+
+A service that simplifies receiving Bitcoin silent payments for wallets and applications.
+
```
Comment (theanmolsharma) [inline README.md:4]:
```suggestion

> [!WARNING]
> This application is currently in an experimental stage and should be used with caution. It has not undergone extensive testing and may contain bugs, vulnerabilities, or unexpected behavior. Mainnet use is strictly NOT recommended.

```

**Review by theanmolsharma — COMMENTED** [review]
Looks good. Thanks for picking it up.
I'd appreciate it very much if you could add the change that I requested. Once that is in, I'll merge the PR.

**Review by theanmolsharma — COMMENTED** [review]
You need to sign the commits before we can merge.
Please squash all commits to one and sign.

**Comment by seekersoftec** [comment]
Done

**Pull request merged**